### ISSUE TITLE
Readily available Python wheels for ARM?

### ISSUE THREAD

**Issue by m-roberts** [microsoft/onnxruntime#8874]
Currently, [PyPI](https://pypi.org/project/onnxruntime/#files) does not support `armv7l` platform for wheels. In response to this, [piwheels](https://www.piwheels.org/) was added to Raspberry Pi OS to offer pre-compiled builds of Python modules. piwheels builds from source in PyPI, which is not available for onnxruntime, and so there is still no widely-accepted place to get a Raspberry Pi-compatible wheel for onnxruntime.

I filed this issue at piwheels: https://github.com/piwheels/packages/issues/202

Is there any way that this can be resolved by providing source to PyPI, or providing ARM wheels in a separate Python repository?

**Comment by snnn** [comment]
Right, there's no source distribution for onnx and onnxruntime. That's the reason. 

Is your Raspberry Pi OS  32-bit or 64-bit? And what's your Raspberry Pi version ? Are armv7l boards still popular?

**Comment by m-roberts** [comment]
So.. most models of the Raspberry Pi are 64-bit, but Raspberry Pi OS itself is still 32-bit and will be for the foreseeable future.
A 64-bit version is available for preview, but is not yet considered stable.

For a while at least, `armv7l` support is necessary for adoption in most Raspberry Pi use-cases - I don't really know about other `armv7l` boards.

Reference:
https://raspberrytips.com/raspberry-pi-os-64-bits-vs-32-bits/

**Comment by snnn** [comment]
Technically, it is all because we can't find a working centos armv7l docker image.  There is one for centos7 but it is not functional. And CentOS8/UBI even refused to support armv7l . So we can't do it like what we are doing for x86_64 and aarch64: build it on CentOS7 and assume it is suitable for most Linux distros.

So the question becomes whether we should provide a special build just for Raspberry Pi OS  32-bit and where to host it? Pypi is not the right place because the wheel won't conform to any manylinux standard. piwheels  seems to be a good choice. But, do you know which  Raspberry Pi OS version and gcc version it uses? 

BTW, it's very easy to cross-compile a binary for C/C++/java/C#/..., but not for python. Python's setuptools doesn't support cross-compiling at all. And I can't get the CentOS7 armv7l docker image running with qemu on x86_64. So I ran out of my methods.

**Comment by m-roberts** [comment]
Would still love to see this!

**Comment by Joldiges** [comment]
> itself is still 32-bit

Me too
